src/components/card/HistoryVideoCard.jsx

```diff
@@ -0,0 +1,76 @@
+import React from "react";
+import styles from "./card.module.css";
+import { Link, useNavigate } from "react-router-dom";
+import {
+  useHistory,
+  useLike,
+  usePlaylist,
+  useUser,
+  useWatchLater,
+} from "context";
+import { DeleteIcon } from "assets/icons/icons";
+import { useApi } from "custom-hooks";
+
+function HistoryVideoCard({ details, extra, type }) {
+  const navigate = useNavigate();
+  const { getToken } = useUser();
+  const {
+    thumbnail,
+    title,
+    url,
+    subscribers,
+    channelName,
+    avatar,
+    description,
+    _id,
+  } = details;
+
+  const { deleteLikedVideoApi, deleteWatchLaterVideoApi } = useApi();
+
+  const { postHistoryApi } = useApi();
+  const { history, deleteFromHistory } = useHistory();
+
+  const { deleteDataUsingApi: deleteWatchLater } = useWatchLater();
+  const { deleteDataUsingApi: deleteLike } = useLike();
+
+  const { deleteFromPlaylistHandler, playlist } = usePlaylist();
+
+  const onClickHandler = () => {
+    if (type === "fromhistory") {
+      deleteFromHistory(_id);
+    } else if (type === "fromwatchlater") {
+      deleteWatchLater(deleteWatchLaterVideoApi, _id);
+    } else if (type === "fromliked") {
+      deleteLike(deleteLikedVideoApi, _id);
+    } else if (type === "fromplaylist") {
+      deleteFromPlaylistHandler(extra._id, details._id, extra.title);
+    }
+  };
```

Comment:
this function can be extracted out, this way u can reuse it in all the other cards as well

Reply:
Yes, i have extracted this function from playlist- context. And using wherever needed !!